modules/core/src/main/java/org/apache/ignite/internal/client/impl/connection/GridClientNioTcpConnection.java

```diff
@@ -846,13 +850,38 @@ private GridClientAuthenticationRequest buildAuthRequest() {
     }
 
     /** {@inheritDoc} */
-    @Override public GridClientFuture<Boolean> currentState(UUID destNodeId)
-        throws GridClientClosedException, GridClientConnectionResetException {
-        GridClientStateRequest msg = new GridClientStateRequest();
+    @Override public GridClientFuture<?> changeState(
+        ClusterState state,
+        UUID destNodeId
+    ) throws GridClientClosedException, GridClientConnectionResetException {
+        assert state != null;
```

Comment:
and what if asserts disabled ? is it would be correct near U.writeEnum(out, state); and after all ?

Reply:
> and what if asserts disabled ?

I've changed to NPE.

> is it would be correct near U.writeEnum(out, state); and after all ?

Not at all. We use one message for request current cluster state and change them.